reo/utilities.py

```diff
@@ -253,35 +253,103 @@ def check_common_outputs(Test, d_calculated, d_expected):
     year_profile_hourly_series = pd.Series(np.zeros(8760), index=dt_profile)
-    for i in range(len(relative_periods_df)):
-        start_month = relative_periods_df["Month"][i]
-        start_week = relative_periods_df["Start Week of Month"][i]
-        start_weekday = relative_periods_df["Start Day of Week"][i]  # Monday is 0, Sunday is 6
-        day_of_month = calendar.Calendar().monthdayscalendar(year=year,month=start_month)[start_week][start_weekday]
-        hour_of_day = relative_periods_df["Start Hour"][i]
-        start_datetime = datetime.datetime(year=year, month=start_month, day=day_of_month, hour=hour_of_day)
-        duration = pd.to_timedelta(relative_periods_df["Duration"][i])
-        year_profile_hourly_series[start_datetime:start_datetime+duration-datetime.timedelta(hours=1)] = 1.0
+    
+    # Check if the consecutive_periods is a list_of_dict or other (must be Pandas DataFrame), and if other, convert to list_of_dict
+    if not isinstance(consecutive_periods, list):
+        consecutive_periods = convert_dataframe_to_list_of_dict(consecutive_periods)
+
+    day_of_week_name = ["Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday", "Sunday"]
+    for i in range(len(consecutive_periods)):
+        start_month = int(consecutive_periods[i]["month"])  # One-indexed both user input and Calendar package
+        start_week_of_month = int(consecutive_periods[i]["start_week_of_month"] - 1)  # One-indexed for user, but zero-index for Calendar
+        start_day_of_week = int(consecutive_periods[i]["start_day_of_week"] - 1)  # Monday - Sunday is 1 - 7 for user, 0 - 6 for Calendar
+        start_hour = int(consecutive_periods[i]["start_hour"] - 1)  # One-indexed for user, datetime hour is zero-index
+        duration_hours = int(consecutive_periods[i]["duration_hours"])
+        error_start_text = "Error in chp_unavailability_period {}. ".format(i+1)
+        try:
+            start_day_of_month = calendar.Calendar().monthdayscalendar(year=year,month=start_month)[start_week_of_month][start_day_of_week]  # One-indexed
+            if start_day_of_month == 0:  # This may happen if there is no day_of_week in the 1st, 5th or 6th week of the month
+                raise DayOfWeekError("There is no start_day_of_week {} ({}) in week {} of month {} in the year {}. Remember, Monday is treated as the first day of the week.".format(start_day_of_week+1, day_of_week_name[start_day_of_week], start_week_of_month+1, start_month, year))
+            else:
+                start_datetime = datetime.datetime(year=year, month=start_month, day=start_day_of_month, hour=start_hour)
+                if start_datetime + datetime.timedelta(hours=duration_hours-1) > dt_profile[-1]:
+                    raise DurationOverflowsYearError("The start day/time and duration_hours exceeds the end of the year. Please specify two separate unavailability periods: one for the beginning of the year and one for up to the end of the year")
```

Comment:
Might add text to say which dict triggered this error

Reply:
The dict/period is appended to the front of the error message in the error_start_text:

>error_start_text = "Error in chp_unavailability_period {}. ".format(i+1)

>except DurationOverflowsYearError as e:
>errors_list.append(error_start_text + str(e.args[0]))